fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RecordConstructorValue.java

```diff
@@ -377,14 +379,43 @@ public Object finish() {
             }
 
             @Nonnull
-            private List<Accumulator> buildAccumulators() {
+            private List<Accumulator> buildAccumulators(@Nullable List<RecordCursorProto.AccumulatorState> initialState) {
+                List<Value> childrenAsList = new ArrayList<>();
+                getChildren().forEach(childrenAsList::add);
                 final ImmutableList.Builder<Accumulator> childAccumulatorsBuilder = ImmutableList.builder();
-                for (final var child : getChildren()) {
-                    Verify.verify(child instanceof AggregateValue);
-                    childAccumulatorsBuilder.add(((AggregateValue)child).createAccumulator(typeRepository));
+
+                if (initialState != null) {
+                    Verify.verify(initialState.size() == childrenAsList.size());
```

Comment:
Can this be false due to some kind of weird composition? I'm thinking something like what if one of the child accumulators is an `avg`, and so it actually produces more state. It would seem like we'd either need to have state's nest, or we'd need to handle the fact that the list can be flattened

Reply:
I added a test `FDBStreamAggregationPlanTest.partialAggregateAggregateThreeGroupByTwo`, where it has 3 aggregator, one of them is `avg`. It works correctly, because for `avg`, `initialState = [state, state]` as we defined it as a repeated field.